types/index.d.ts

```diff
@@ -15,9 +17,9 @@ export type Diff<T extends string, U extends string> = (
 /**
  * Drop keys `K` from `T`.
  *
- * @see https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-307871458
+ * @see https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-377567046
  */
-export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>;
+export type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>;
 
 /**
  * Find the overlapping variants between two string unions.
```

Comment:
Can we replace `Overlap` with `Extract`?

Reply:
I'm not familiar with `Overlap` or the internals of `Extract` yet, happy to do whatever you suggest!

Reply:
```ts
/**
 * Extract from T those types that are assignable to U
 */
type Extract<T, U> = T extends U ? T : never;
```

`Diff` and `Overlap` are dual, as `Exclude` and `Extract` are dual.

Reply:
Cool! So what's the specific action you would like here? Sounds like the answer to your first question is "Yes", but do you want me to just delete the `Overlap` implementation from type-zoo?

Reply:
Yes, I think `Diff` and `Overlap` have been superseded by `Exclude` and `Extract` respectively.

Reply:
So let's delete both `Diff` and `Overlap`.

Reply:
Deleted!